src/databricks/labs/ucx/source_code/files.py

```diff
@@ -44,12 +48,79 @@ def __repr__(self):
         return f"<LocalFile {self._path}>"
 
 
+class LocalDirectory(SourceContainer):
+    def __init__(self, path: Path, file_loader: FileLoader, dir_loader: DirectoryLoader):
+        self._path = path
+        self._file_loader = file_loader
+        self._dir_loader = dir_loader
+
+    @property
+    def path(self) -> Path:
+        return self._path
+
+    def build_dependency_graph(self, parent: DependencyGraph) -> list[DependencyProblem]:
+        return list(self._build_dependency_graph(parent))
+
+    def _build_dependency_graph(self, parent: DependencyGraph) -> Iterable[DependencyProblem]:
+        for child_path in self._path.iterdir():
+            if child_path.is_dir():
+                dependency = Dependency(self._dir_loader, child_path)
+                yield from parent.register_dependency(dependency).problems
+                continue
```

Comment:
I prefer a `if-else` here, but this pattern is also common in this code base

Reply:
or a `loader = self._dir_loader if child_path.is_dir() else self._file_loader` with one `dependency = Dependency(loader, child_path); yield from parent.register_dependency(dependency).problems`

Reply:
Yup